### DISCUSSION TITLE
Mobile apps and React native plan

### DISCUSSION THREAD

**Discussion by karimsx** [blitz-js/blitz#1926] [General] [answered]
### What do you want and why?

Hey everyone, i want to know when it is planned to have react-native support, i'm planning to start a new app, i will be working on the backend, waiting for it to be implemented.

and thanks you for this awesome framework, its is an genuise idea to encapsulate the api/graphql call layers, that permit a seamless full stack integration.

### Possible implementation(s)

support react native

**Comment by AndriusBartulis** [comment]
I am also interested in this. I wonder if it's worth having a wider conversation of the best mobile story in the context of fullstack and Blitz. Is react native the right approach for the smaller teams/projects that Blitz audience mainly work on? What about Capacitor based hybrid apps that would allow wrapping pretty much all web React components/pages, do a bit of processing, and output a hybrid app wrapped in Capacitor, ready to be built for iOS and Android. Interested to see what ideas others have about this.

**Comment by karimsx** [comment]
> I am also interested in this. I wonder if it's worth having a wider conversation of the best mobile story in the context of fullstack and Blitz. Is react native the right approach for the smaller teams/projects that Blitz audience mainly work on? What about Capacitor based hybrid apps that would allow wrapping pretty much all web React components/pages, do a bit of processing, and output a hybrid app wrapped in Capacitor, ready to be built for iOS and Android. Interested to see what ideas others have about this.

and would be very interesting to have ionic v5 as it already support react

**Comment by AndriusBartulis** [comment]
> 
> 
> > I am also interested in this. I wonder if it's worth having a wider conversation of the best mobile story in the context of fullstack and Blitz. Is react native the right approach for the smaller teams/projects that Blitz audience mainly work on? What about Capacitor based hybrid apps that would allow wrapping pretty much all web React components/pages, do a bit of processing, and output a hybrid app wrapped in Capacitor, ready to be built for iOS and Android. Interested to see what ideas others have about this.
> 
> and would be very interesting to have ionic v5 as it already support react

Yup. Ionic could work. I guess the question is whether most people building apps in small teams want to rewrite their web app in React Native, or whether they are happy give up some of React Native performance and features and instead get significant time savings by being able to reuse a large percentage of their existing React app code.

The area I am most excited about is automating the extraction of blitz pages and components and generation of a fully functional mobile app from it, with almost no interaction from developer (unless if they want to customise the mobile experience of course).

Though I am aware a lot of people prefer React Native over hybrid approaches like Ionic. Looking forward to hear more thoughts.

**Comment by karimsx** [comment]
> The area I am most excited about is automating the extraction of blitz pages and components and generation of a fully functional mobile app from it, with almost no interaction from developer (unless if they want to customise the mobile experience of course).

using some ui library like material-ui and some responsive design can make application have a great mobile experience too, and seamless integration between web and mobile component ?

**Comment by flybayer** [comment, marked as answer]
### Here's the plan

**Currently:**

- You have access to the generated API endpoints and you can use them in any mobile app (we currently have a few people doing this)
- You can see all generated endpoints using [`blitz routes`](https://blitzjs.com/docs/cli-routes) command
- You can see the RPC specification for the endpoints here: https://blitzjs.com/docs/rpc-specification

**What we plan to do going forward:**

1. First step, add docs and tools for making it easier to manually use the generated API endpoints. See https://github.com/blitz-js/blitz/discussions/1907 and https://github.com/blitz-js/blitz/issues/745
2. Next, add ability generate a JS/TS client library for easily using the generated endpoints from an app
3. Next, try to add first-class react-native support where you can import blitz queries/mutations directly into your react-native code and have it magically work just like it does today with your web app

**Comment by BjoernRave** [comment]
Hey guys, I created this repo, which shows hwo I setup react native with auth on blitz: https://github.com/BjoernRave/Blitz-React-Native-Auth
